meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -97,7 +97,13 @@ public LeaderElectionClient(MetaClientConfig metaClientConfig, String participan
     LOG.info("Creating MetaClient for LeaderElectionClient");
     if (MetaClientConfig.StoreType.ZOOKEEPER.equals(metaClientConfig.getStoreType())) {
       ZkMetaClientConfig zkMetaClientConfig = new ZkMetaClientConfig.ZkMetaClientConfigBuilder().setConnectionAddress(
-          metaClientConfig.getConnectionAddress()).setZkSerializer((new LeaderInfoSerializer())).build();
+              metaClientConfig.getConnectionAddress())
+          .setZkSerializer((new LeaderInfoSerializer()))
+          .setSessionTimeoutInMillis(metaClientConfig.getSessionTimeoutInMillis())
+          .setMetaClientReconnectPolicy(metaClientConfig.getMetaClientReconnectPolicy())
+          .setConnectionInitTimeoutInMillis(metaClientConfig.getConnectionInitTimeoutInMillis())
+          .setAuthEnabled(metaClientConfig.isAuthEnabled())
+          .build();
```

Comment:
Why we don't have config validator? Or these are the optional fields.

Reply:
These are optional fields. If user do not populate, they will have default values.